### ISSUE TITLE
ldap_anonymous_bind parameter is the reverse of what is expected

### ISSUE THREAD

**Issue by draeath** [cobbler/cobbler#2941]
### Describe the bug

When configuring `authentication.ldap` one sees the following parameters in Cobbler 3.2:

    ldap_anonymous_bind: false
    ldap_search_bind_dn: ''
    ldap_search_passwd: ''

The meaning of `ldap_anonymous_bind` is backwards. An anonymous bind is done without authentication. If you set this boolean to `true` the two following parameters (bind username/password) are used and an anonymous search is NOT attempted. Setting this to `false` causes an anonymous search to occur.

### Steps to reproduce

1. fresh cobbler 3.2 install on RHEL 8 via EPEL (though CentOS 8 would have the same behavior?)
2. configure `authentication.ldap` in `modules.conf` and LDAP settings in `settings.yaml`
3. utilize plain LDAP or use a stunnel proxy to expose the connection traffic
4. inspect the LDAP transaction via tcpdump or wireshark

### Expected behavior

- `ldap_anonymous_bind: true` should cause anonymous (unauthenticated) LDAP search behavior.
- `ldap_anonymous_bind: false` should cause an authentication (bind) attempt for the search.

### Cobbler version

````
Cobbler 3.2.2
  source: ?, ?
  build time: Mon Jan 24 02:22:30 2022
````

### Operating system

RHEL 8.5

### Cobbler log

````
N/A
````

### Screenshots

N/A

### Additional information

    # cat /etc/redhat-release 
        Red Hat Enterprise Linux release 8.5 (Ootpa)

    # dnf info cobbler cobbler-web
        Updating Subscription Management repositories.
        Last metadata expiration check: 0:44:37 ago on Tue 15 Feb 2022 02:57:37 PM EST.
        Installed Packages
        Name         : cobbler
        Version      : 3.2.2
        Release      : 8.module_el8+13675+f0fb7535
        Architecture : noarch
        Size         : 2.4 M
        Source       : cobbler-3.2.2-8.module_el8+13675+f0fb7535.src.rpm
        Repository   : @System
        From repo    : epel-modular
        Summary      : Boot server configurator
        URL          : https://cobbler.github.io/
        License      : GPLv2+
        Description  : Cobbler is a network install server.  Cobbler supports PXE, ISO
                     : virtualized installs, and re-installing existing Linux machines.
                     : The last two modes use a helper tool, 'koan', that integrates with
                     : cobbler.  There is also a web interface 'cobbler-web'.  Cobbler's
                     : advanced features include importing distributions from DVDs and rsync
                     : mirrors, kickstart templating, integrated yum mirroring, and built-in
                     : DHCP/DNS Management.  Cobbler has a XML-RPC API for integration with
                     : other applications.

        Name         : cobbler-web
        Version      : 3.2.2
        Release      : 8.module_el8+13675+f0fb7535
        Architecture : noarch
        Size         : 1.3 M
        Source       : cobbler-3.2.2-8.module_el8+13675+f0fb7535.src.rpm
        Repository   : @System
        From repo    : epel-modular
        Summary      : Web interface for Cobbler
        URL          : https://cobbler.github.io/
        License      : GPLv2+
        Description  : Web interface for Cobbler that allows visiting
                     : http://server/cobbler_web to configure the install server.

**Comment by draeath** [comment]
I am an idiot and it took until just now to realize it.

Please disregard.

**Comment by draeath** [comment]
Reopening. I had confused myself while writing this. I actually do want this open.

Relevant code snippet from `site-packages/cobbler/modules/authentication/ldap.py`:

```python
    # if we're not allowed to search anonymously, grok the search bind settings and attempt to bind
    if api_handle.settings().ldap_anonymous_bind:
        searchdn = api_handle.settings().ldap_search_bind_dn
        searchpw = api_handle.settings().ldap_search_passwd

        if searchdn == '' or searchpw == '':
            raise CX("Missing search bind settings")
```

**Comment by SchoolGuy** [comment]
@draeath Wouldn't it be even easier to just use the credentials if they are given and otherwise not? The default settings for this are empty and thus existing configurations are not broken. Is this something you would deem straightforward?

**Comment by draeath** [comment]
Yes, that would do. At least if it was documented... Perhaps in a comment in the default file?

**Comment by SchoolGuy** [comment]
The problem is due to the automigrations the comments will vanish. Thus things need to be either self explanatory or it won't work. So I guess we need to have a breaking change in either way. I will think about this once I have fixed some more bugs and have time for this.
